### PR TITLE
Issue91 new type cli

### PR DESCRIPTION
**Pull request by DannyPeck**
- Refactored the new command cli to include a type argument with the possible values of "bin", "static", or "shared".
- Cleaned up the implementation of the new command and added progress print outs.

### CODE REVIEW

**Comment by DannyPeck** [comment]
This PR covers both #91 and #93.

**Review by xmclark — DISMISSED** [review]
Nice!

**Review by technetos — DISMISSED** [review]

File: hatch/src/cli/commands/new.rs [hunk 1]
```diff
@@ -39,16 +39,14 @@ impl<'new> New {
...
-        _ => ProjectKind::Library(LibraryKind::Static),
-      };
-
-      kind
+      // we cannot use the static variables BIN, STATIC, or SHARED directly because it is illegal in
+      // Rust to pattern match on a static variable
+      match type_arg.as_str() {
+        value if (value == BIN) => ProjectKind::Binary,
+        value if (value == STATIC) => ProjectKind::Library(LibraryKind::Static),
+        value if (value == SHARED) => ProjectKind::Library(LibraryKind::Shared),
```
Comment (xmclark) [inline hatch/src/cli/commands/new.rs]: Are the parenthesis required in `value if (value == FOO) => ... `? If they are not, I think we should omit them. I like the more terse syntax here. I'm open to counters.

Comment (DannyPeck) [reply hatch/src/cli/commands/new.rs]: I personally am a fan of the parenthesis.  I think that with the arrow syntax right after is just too much.  This leads to better visualization of the semantics of the statement.

Comment (technetos) [reply hatch/src/cli/commands/new.rs]: Oh lets get rid of them!

**Review by xmclark — APPROVED** [review]

**Review by technetos — APPROVED** [review]

**Review by xmclark — APPROVED** [review]

**Pull request merged**